src/main/java/duke/ui/Parser.java

```diff
@@ -0,0 +1,207 @@
+        String[] inputs = s.split(" ");
+        String[] acceptableInputs = new String[]
+            {"mark", "unmark", "todo", "deadline", "event", "delete", "find"};
+        boolean isAcceptable = false;
+        for (String acceptableInput : acceptableInputs) {
+            if (acceptableInput.equals(inputs[0])) {
+                isAcceptable = true;
+                break;
+            }
+        }
+        if (!isAcceptable) {
+            throw new WrongInputException("D: D: D: I don't understand the input D: D: D:");
+        }
+        return inputs[0];
+    }
+
+    /**
+     * Returns the numerical part of the user input.
+     * Assumes the user input contains a command that requires a numerical input.
+     *
+     * @param s The user input.
+     * @param command The command contained in the user input.
+     * @param length The length of the task list.
+     * @return The numerical part of the user input.
+     * @throws WrongInputException If the input is not in the expected form.
+     */
+    public int parseNumericalDescription(String s, String command, int length) throws WrongInputException {
+        String[] inputs = s.split(command + " ");
+        if (length == 0) {
+            throw new WrongInputException("D: D: D: There are no items in the list D: D: D:");
+        }
+        if (inputs.length > 2) {
+            throw new WrongInputException("D: D: D: The " + command
+                    + " should be followed by a single number D: D: D:");
+        }
+
+        try {
+            int tasknumber = Integer.parseInt(inputs[1]);
+            if (tasknumber > length || tasknumber < 1) {
+                throw new WrongInputException("D: D: D: The task number isn't in the list D: D: D:");
+            }
+            return tasknumber - 1;
+        } catch (NumberFormatException e) {
+            throw new WrongInputException("D: D: D: The input doesn't seem to contain a number D: D: D:");
+        }
+    }
+
+    /**
+     * Returns an array containing the task description and its associated time.
+     *
+     * @param s The user input.
+     * @param command The command contained in the user input.
+     * @param format The regular expression that indicates the timing part of the task.
+     * @return An array containing the task description and its associated time.
+     * @throws WrongInputException If the regular expression cannot be found in the user input.
+     * @throws IncompleteInputException If the input is incomplete.
+     */
+    public String[] parseFormatDescription(String s, String command, String format)
+            throws WrongInputException, IncompleteInputException {
+        try {
+            String[] inputs = s.split(" " + format + " ");
+            if (inputs.length < 2) {
+                throw new IncompleteInputException("D: D: D: The description of a " + command
+                        + " is incorrect or empty D: D: D:");
```

Comment:
I like the 8 spaces for the indentation for the wrapped line. Great adherence to the coding standard.